src/test/testRunner.ts

```diff
@@ -0,0 +1,29 @@
+/*!
+ * Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as legacyTestRunner from '../../third-party/test/index'
+
+export interface RunTestsParameters {
+    /**
+     * The root folder containing all tests to be run
+     */
+    rootTestsPath: string
+}
+
+/**
+ * Utility method to invoke tests. Abstracts away the test framework (currently mocha)
+ */
+export async function runTests(parameters: RunTestsParameters): Promise<void> {
+    return new Promise<void>((resolve, reject) => {
+        legacyTestRunner.runTests(parameters.rootTestsPath, (err: any, failureCount: number) => {
+            if (err) {
+                reject(err)
+            }
+
+            console.log(`Tests completed with ${failureCount} failure(s)`)
```

Comment:
This needs to reject if there are failures, else the test run is considered a success. [Example](https://travis-ci.org/aws/aws-toolkit-vscode/jobs/580324521)